src/Support/SupportRESTController.php

```diff
@@ -101,10 +103,10 @@ public function callback( WP_REST_Request $request ) {
 		$request_args = $request->get_param( 'args' );
 		$request_args = ( ! empty( $request_args ) && is_array( $request_args ) ) ? $request_args : [];
 
-		$this->support_data->set_args( $request_args );
-		$support_response = $this->support_data->send_data();
+		$support_data     = $this->injector->make( SupportData::class, [ 'args' => $request_args ] );
```

Comment:
Same suggestion as above, creating the `SupportData` object should be simple enough.

```suggestion
		$support_data = new SupportData( $args );
```

Reply:
Originally it was `$support_data = new SupportData( $args );`. I have used an injector after suggestions from the previous PR.

Reply:
Hmm yea, but unless I'm missing something there's no benefit in creating the `SupportData` via the injector as its not a service or anything. cc @westonruter.

Reply:
Another example of using the injector to create a class can be found in `PairedRouting`:

https://github.com/ampproject/amp-wp/blob/002f88e8096a8870c666244577d533311728804f/src/PairedRouting.php#L252-L263

While in practice it makes no difference, in theory the injector is better because this could reference an interface instead of an actual class, and the bindings defined in `AmpWpPlugin` could automatically route the class to instantiate. This could make it easier to test with a mock, as I understand. 

Also, by using `make` the injector is able to re-use instances which are listed among the shared instances in `AmpWpPlugin`, which isn't possible otherwise.

So yeah, I understand from dependency injection that classes should generally not be instantiated directly. 

cc @schlessera

Reply:
Thanks for the explanation @westonruter. In this scenario we wouldn't be benefiting from using the injector, since the class being instantiated does not require any sort of dependency resolution and so would not make use of the bindings or shared instances provided by `AmpWpPlugin`. However,  I do understand the benefits of using the injector with regards to testing especially, so after giving it more thought I'm OK with the current implementation.